src/main/java/io/airlift/units/Count.java

```diff
@@ -0,0 +1,200 @@
+
+        this.value = size;
+        this.unit = unit;
+    }
+
+    public double getValue()
+    {
+        return value;
+    }
+
+    public Unit getUnit()
+    {
+        return unit;
+    }
+
+    public double getValue(Unit unit)
+    {
+        return value * (this.unit.getFactor() * 1.0 / unit.getFactor());
+    }
+
+    public long roundTo(Unit unit)
+    {
+        double rounded = Math.floor(getValue(unit) + 0.5d);
+        checkArgument(rounded <= Long.MAX_VALUE,
+                "size is too large to be represented in requested unit as a long");
+        return (long) rounded;
+    }
+
+    public Count convertTo(Unit unit)
+    {
+        requireNonNull(unit, "unit is null");
+        return new Count(getValue(unit), unit);
+    }
+
+    public Count convertToMostSuccinctCount()
+    {
+        Unit unitToUse = Unit.SINGLE;
+        for (Unit unitToTest : COUNT_UNITS) {
+            if (getValue(unitToTest) > 0.999_999) {
+                unitToUse = unitToTest;
+            }
+            else {
+                break;
+            }
+        }
+        return convertTo(unitToUse);
+    }
+
+    @JsonValue
+    @Override
+    public String toString()
+    {
+        //noinspection FloatingPointEquality
+        if (floor(value) == value) {
+            return (long) (floor(value)) + unit.getUnitString();
+        }
+
+        return String.format(Locale.ENGLISH, "%.2f%s", value, unit.getUnitString());
+    }
+
+    @JsonCreator
+    public static Count valueOf(String count)
+            throws IllegalArgumentException
+    {
+        requireNonNull(count, "count is null");
+        checkArgument(!count.isEmpty(), "count is empty");
+
+        Matcher matcher = PATTERN.matcher(count);
+        if (!matcher.matches()) {
+            throw new IllegalArgumentException("Not a valid count string: " + count);
+        }
+
+        double value = Double.parseDouble(matcher.group(1));
+        String unitString = matcher.group(2);
+
+        for (Unit unit : Unit.values()) {
+            if (unit.getUnitString().equals(unitString)) {
+                return new Count(value, unit);
+            }
+        }
+
+        throw new IllegalArgumentException("Unknown unit: " + unitString);
+    }
+
+    @Override
+    public int compareTo(Count o)
+    {
+        return Double.compare(getValue(Unit.SINGLE), o.getValue(Unit.SINGLE));
+    }
+
+    @Override
+    public boolean equals(Object o)
+    {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+
+        Count dataSize = (Count) o;
+
+        return compareTo(dataSize) == 0;
+    }
+
+    @Override
+    public int hashCode()
+    {
+        double value = getValue(Unit.SINGLE);
+        return Double.hashCode(value);
+    }
+
+    public enum Unit
+    {
+        //This order is important, it should be in increasing magnitude.
+        SINGLE(1L, ""),
+        THOUSAND(1000L, "k"),
```

Comment:
Do we want to use a lowercase `k` here?

Reply:
I took from here: https://en.wikipedia.org/wiki/Power_of_10

They mention both ways. Changed.